src/CGIHandler.cpp

```diff
@@ -12,10 +10,110 @@ CGIHandler::~CGIHandler()
 {
 }
 
+CGIHandler &CGIHandler::operator=(const CGIHandler &other)
+{
+	if (this != &other)
+	{									   // Protect against self-assignment
+		ARequestHandler::operator=(other); // Call the base class assignment operator
+										   // Copy or assign other members of CGIHandler if necessary
+	}
+	return *this;
+}
+
 HTTPResponse CGIHandler::handleRequest(const HTTPRequest &request)
 {
-	(void)request;
+	std::string cgiOutput = handleCGIRequest(request);
+	// You'll need to construct an HTTPResponse object from cgiOutput.
+	// This is a simplified example. You should set the appropriate headers and status code.
 	HTTPResponse response;
-	// Determine CGI script path and set up environment
+	response.setBody(cgiOutput);
 	return response;
-}
\ No newline at end of file
+}
+
+std::string CGIHandler::handleCGIRequest(const HTTPRequest &request)
+{
+	Environment env;
+
+	// load the meta vars from the request to env
+	env.HTTPRequestToMetaVars(request, env);
+	std::cout << env;
+
+	std::string cgiOutput = executeCGI(env);
+
+	return cgiOutput;
+}
+
+// NOTE FOR SELF: key1=value1&key2=value2&key3=value3 might not be directly passed as command-line arguments (i.e., in
+// argv) it migth  be passed to the script as part of the environment variables
+//=> what is passed as command-line arguments ? path to the script + the path to the file to process ?
+char *const *CGIHandler::createArgvForExecve(const Environment &env)
+{
+	char **argv = new char *[2];
+
+	std::string scriptName = env.getVar("SCRIPT_NAME");
+	std::string pathTranslated = env.getVar("PATH_TRANSLATED");
+	std::string scriptPath = pathTranslated + scriptName;
+	std::cout << std::endl << "\033[31mscriptPath: " << scriptPath << "\033[0m" << std::endl;
+
+	argv[0] = new char[scriptPath.length() + 1]; // +1 for the null terminator
+	ft_strcpy(argv[0], scriptPath.c_str());
+
+	argv[1] = NULL;
+
+	return argv;
+}
+
+std::string CGIHandler::executeCGI(const Environment &env)
+{
+
+	char *const *argv = createArgvForExecve(env);
+
+	int pipeFD[2];
+	if (pipe(pipeFD) == -1)
+	{
+		perror("pipe failed");
+		_exit(EXIT_FAILURE);
+	}
+
+	pid_t pid = fork();
+	if (pid == -1)
+	{
+		perror("fork failed");
+		_exit(EXIT_FAILURE);
```

Comment:
What is this _exit function with underscore?

Reply:
Was accidental but:

> 
> _exit() won't flushes the stdio buffer while exit() flushes the stdio buffer prior to exit.
> 
> _exit() can not perform clean-up process while exit() can be registered with some function ( i.e on_exit or at_exit) to perform some clean-up process if anything is required before existing the program.
> 

> In the child branch of a fork(), it is normally incorrect to use exit(), because that can lead to stdio buffers being flushed twice, and temporary files being unexpectedly removed.


Copied from:
https://stackoverflow.com/questions/5422831/what-is-the-difference-between-using-exit-exit-in-a-conventional-linux-fo#:~:text=_exit()%20won't%20flushes,required%20before%20existing%20the%20program.

Reply:
Good to know! Neither `exit` not `_exit` seems to be allowed by the subject. Edit: both exit and _exit are present in the cstdlib header, which is a C++ header and so we can use them. In any case if a CGI is failing we don't want to shut down the server. We don't need to solve this in this branch, but maybe open an issue or add a note on the exiting issue about the exit(s) and maybe find out what would be the most appropriate behavior in these tow cases.

Reply:
The one you commented about I already replaced, but I will get more info about the use of _exit() for the fork